neuro_flow/expr.py

```diff
@@ -1346,7 +1363,25 @@ def __hash__(self) -> int:
         return hash((self.__class__.__name__, self._items))
 
 
-class MappingExpr(StrictExpr[MappingT]):
+class BaseMappingExpr(BaseExpr[MappingT]):
+    pass
+
+
+class MergeMappingsExpr(BaseMappingExpr):
+    def __init__(self, *mappings: BaseMappingExpr):
+        self._mappings = mappings
+
+    async def eval(self, root: RootABC) -> MappingT:
+        ret = {}
+        for mapping in reversed(self._mappings):
+            value = await mapping.eval(root)
+            if value:
```

Comment:
YAML has a special notation for None (`~` of missing value in a mapping) to distinguish from an empty string.
Should we replace the check with `if value is None:` to explicitly allow empty strings and zeroes as *inheritance termination* markers?
The same question for scalar merging.

Reply:
Actually, this `if` is not really useful - it needed because of types. The `BaseMappingExpr` inherits from `BaseExpr[MappingT]` that defines `eval` return type as `Optional[MappingT]`, but all real descendants of `BaseMappingExpr` never return `None` from eval. I think I can tweak types to avoid this.

The YAMLs `~` is handled in other places using `_specified_fields` that stores what fields were defined in yaml.

Reply:
Thanks for the explanation